docs: document AI-Q 2.2 candidate scope (#322)
NVIDIA-AI-Blueprints/aiq-research-assistant: docs/source/customization/guardrails.md

AI-Q can use NeMo Guardrails through NeMo Agent Toolkit middleware to evaluate selected workflow and agent-boundary inputs and outputs. Guardrails can pass content through unchanged, block content with a configured refusal, or modify selected fields before execution continues.

AI-Q provides middleware types for workflow, shallow-research, and deep-research boundaries. That capability does not
mean every boundary is active whenever its middleware is defined. In
`configs/config_web_default_guardrails.yml`, the workflow middleware is explicitly attached to the workflow. The async
deep-research runner selects `deep_agent_guardrails` because its `workflow_functions` targets `deep_research_agent`.
The checked-in profile does **not** attach `shallow_agent_guardrails` to `shallow_research_agent`, so shallow guardrails
are not active until that middleware reference is added to the function.

## Guarded Boundaries

| Boundary | Middleware | Applies To |
| --- | --- | --- |
| Workflow | `workflow_guardrails` | Workflow input and final assistant response; active when attached under `workflow.middleware`. |

> | Shallow researcher | `shallow_agent_guardrails` | Shallow input/output message content; active when attached under `functions.shallow_research_agent.middleware`. |

Reviewer: ```suggestion
| Shallow researcher | `shallow_agent_guardrails` | Shallow input or output message content; active when attached under `functions.shallow_research_agent.middleware`. |
```

> | Deep researcher | `deep_agent_guardrails` | Deep input/output message content; the async runner selects it when `workflow_functions` targets `deep_research_agent`. |

Reviewer: ```suggestion
| Deep researcher | `deep_agent_guardrails` | Deep input or output message content; the async runner selects it when `workflow_functions` targets `deep_research_agent`. |
```

| Modify | Replace the selected input or output field with the modified content returned by the rail. |
| Block | Return the configured refusal response instead of continuing with the blocked content. |

Input-rail evaluation exceptions are caught, logged, and converted to the middleware refusal response. Output-rail
evaluation exceptions are not converted to a refusal; they propagate and fail the invocation.

## Configuration Shape

The guardrails configuration is placed in the top-level `middleware` section. Defining an entry makes that middleware
available; attach it to the workflow or function that should be guarded. The `guardrails` block uses NAT/NeMo

> Guardrails configuration. See `configs/config_web_default_guardrails.yml` for the full field-selection paths used by

Reviewer: ```suggestion
Guardrails configuration. Refer to `configs/config_web_default_guardrails.yml` for the full field-selection paths used by
```
